effekt/shared/src/main/scala/effekt/RecursiveDescent.scala

```diff
@@ -1128,18 +1128,35 @@ class RecursiveDescent(positions: Positions, tokens: Seq[Token], source: Source)
         case Many(xs, _) => Call(IdTarget(IdRef(List("effekt"), s"Tuple${xs.size}", span().synthesized)), Nil, xs.toList, Nil, span().synthesized)
       }
 
-  def isHole: Boolean = peek(`<>`) || peek(`<{`)
+  def isHole: Boolean = peek.kind match {
+    case `<>` => true
+    case HoleStr(_) => true
+    case _ => false
+  }
   def hole(): Term = {
     nonterminal:
       peek.kind match {
-        case `<>` => `<>` ~> Hole(IdDef("hole", span().synthesized), Return(UnitLit(span().synthesized), span().synthesized), span())
-        case `<{` =>
-          val s = `<{` ~> stmts() <~ `}>`
-          Hole(IdDef("hole", span().synthesized), s, span())
+        case `<>` => `<>` ~> Hole(IdDef("hole", span().synthesized), Template(Nil, Nil), span())
+        case _: HoleStr => {
+          val body = holeTemplate()
+          Hole(IdDef("hole", span().synthesized), body, span())
+        }
         case k => fail("hole", k)
       }
     }
 
+  def holeTemplate(): Template[Stmt] =
+    nonterminal:
+      val first = holeString()
+      val (s, strs) = manyWhile((`${` ~> stmts() <~ `}`, holeString()), `${`).unzip
+      Template(first :: strs, s)
```

Comment:
Can you use `labelled "hole"` here for better error messages?

Reply:
Looked into this, I don't think this will ever be displayed due to how the parser is guarded with `isHole`. We can add it later if there is an example where it would be shown.